code/game/atoms.dm

```diff
@@ -505,6 +507,10 @@
 		if(length(new_overlays))
 			managed_overlays = new_overlays
 			add_overlay(new_overlays)
+		postSignal = TRUE
+
+	if(postSignal)
+		SEND_SIGNAL(src, COMSIG_ATOM_UPDATED_ICON)

```

Comment:
```suggestion
		SEND_SIGNAL(src, COMSIG_ATOM_UPDATED_ICON, signalOut)
```
Also honestly you probably should call on this signal regardless of if changes happened